src/pages/creators/[creator].tsx

```diff
@@ -272,34 +271,49 @@ const CreatorShow: NextPage<CreatorPageProps> = ({ marketplace, creator }) => {
     fetchPolicy: 'network-only',
     variables: {
       creators: [router.query.creator],
-      auctionHouses: [marketplace.auctionHouse.address],
+      auctionHouses: auctionHouses,
       offset: 0,
       limit: 24,
     },
   })
-
   const collectionQuery = useQuery<GetCollectionInfo>(GET_COLLECTION_INFO, {
     variables: {
       creator: router.query.creator,
-      auctionHouses: [marketplace.auctionHouse.address],
+      auctionHouses: auctionHouses,
     },
   })
 
   const nftCountsQuery = useQuery<GetNftCounts>(GET_NFT_COUNTS, {
     fetchPolicy: 'network-only',
     variables: {
       creators: [router.query.creator],
-      auctionHouses: [marketplace.auctionHouse.address],
+      auctionHouses: auctionHouses,
     },
   })
 
+  const listedCountQueryMap = new Map<
+    String,
+    QueryResult<GetNftCounts, OperationVariables>
+  >()
+
+  marketplace.auctionHouses?.forEach((auctionHouse) => {
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    const query = useQuery<GetNftCounts>(GET_LISTED_TOKEN_NFT_COUNT, {
+      variables: {
+        creators: [router.query.creator],
+        auctionHouse: auctionHouse.address,
+      },
+    })
+    listedCountQueryMap.set(auctionHouse.treasuryMint, query)
+  })
```

Comment:
You don't want to do this. hooks should just be called in the render block. Can you provide more info on what is going on here so we can figure out an alternative.

if you do want to manage the query by yourself then get the graphql client from the provider and do a promise all to lookup all the results within a useEffect.

Reply:
We require to find listed nft counts for each individual spl-token. 

If we provide all the auctionHouses within a single query, it will just fetch the total count, but we require individual counts.

Maybe we can have a separate query for this?